pkg/ddevapp/traefik.go

```diff
@@ -409,6 +409,78 @@ func configureTraefikForApp(app *DdevApp) error {
 		}
 	}
 
+	/* The following section reads the project/.ddev/traefik/dynamic_config.*.yaml files, fills any template placeholders in them with the
+	App's templateData (for	targeting the appropriate routers (e.g. {projectname}-web-80-http) or for rewriting the App's URL in the
+	response body), then merges their content into the base dynamic config YAML generated above, and is finally written to /project/.ddev/
+	traefik/config/<project>.yaml. Allows for adding middlewares, overriding settings, etc...
+	*/
+
+	extraDynamicConfigFiles, err := fileutil.GlobFilenames(projectTraefikDir, "dynamic_config.*.yaml")
```

Comment:
All of this processing must happen *before* the traefik `<project>.yaml` is created, or at least before it's populated. Probably before it's created. 

* Template the base config into a string
* Find the dynamic config files if they exist
* util.MergeYamlFile() into the string
* Write it out